server.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+import argparse
+import glob
+import json
+import logging
+import os
+import sys
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from pathlib import Path
+
+import yaml
+from jinja2 import Template
+
+TARGET_NAMESPACE_LABEL = os.getenv("LABEL", "user.kubeflow.org/enabled")
+TEMPLATES_FOLDER = os.getenv("TEMPLATES_FOLDER", "./resources")
+PORT = os.getenv("PORT", 80)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+handler = logging.StreamHandler(sys.stdout)
+handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+def main(port: int, label: str, folder: str) -> None:
+    """
+    Runs a server for injecting resources into namespaces by given label
+    """
+    logger.info("Resource dyspatcher service alive")
+    server = server_factory(port, label, folder)
+    logger.info("Serving forever")
+    server.serve_forever()
+
+
+def server_factory(controller_port: int, label: str, folder: str, url: str = "") -> HTTPServer:
+    """
+    Returns an HTTPServer populated with Handler with customized settings
+    """
+
+    class Controller(BaseHTTPRequestHandler):
+        def sync(self, parent, children):
+            logger.info("Got new request")
+            context = {"namespace": parent.get("metadata", {}).get("name")}
+            pipeline_enabled = parent.get("metadata", {}).get("labels", {}).get(label)
+
+            if pipeline_enabled != "true":
+                logger.info(
+                    f"Namespace not in scope, no action taken (metadata.labels.{label} = {pipeline_enabled}, must be 'true')"  # noqa: E501
+                )
+                return {"status": {}, "children": []}
+
+            desired_secret_count = 0
```

Comment:
`desired_secret_count = 0` says "my desired state is to have no secrets" - is this the intent?  I think this is going to tell metacontroller "if you have 0 secrets, dont ever change that" and then no items will ever be created.  

This probably should be set dynamically based on your manifests.  The upstream sync.py you're using as a guide could get away with a static value here because they knew what it would be, but in your case you do not